From 869474b73f0964a42c305565934b2e101e56db77 Mon Sep 17 00:00:00 2001
From: Martin Roth <martinroth@google.com>
Date: Wed, 24 Feb 2016 13:47:46 -0800
Subject: Memtest86+: Apply crash fix patch

Patch taken from fedora updates:
    http://pkgs.fedoraproject.org/cgit/rpms/memtest86+.git/log/

This is credited to David McInnis, but I couldn't find the
origin of the patch.

Change-Id: I720bd0dda76bc79b43474caa3905c8901c59f828
Signed-off-by: Martin Roth <martinroth@google.com>
Reviewed-on: https://review.coreboot.org/13825
Reviewed-by: Ben Gardner <gardner.ben@gmail.com>
---
 controller.c | 8 +++++---
 init.c       | 3 ++-
 main.c       | 4 ++--
 3 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/controller.c b/controller.c
index 09b40a4..183e9c3 100644
--- a/controller.c
+++ b/controller.c
@@ -2,6 +2,8 @@
  * MemTest86+ V5 Specific code (GPL V2.0)
  * By Samuel DEMEULEMEESTER, sdemeule@memtest.org
  * http://www.canardpc.com - http://www.memtest.org
+ *
+ * Edited by David McInnis Oct 4, 2014
  */
 
 //#include "defs.h"
@@ -1424,7 +1426,7 @@ static void poll_fsb_amd64(void) {
 	unsigned long dramchr;
 	float clockratio;
 	double dramclock;
-	unsigned int dummy[3];
+	unsigned int dummy[4];
 	int ram_type;
 
 	float coef = 10;
@@ -2851,13 +2853,13 @@ static void poll_timings_nf4ie(void) {
 
 static void poll_timings_i875(void) {
 
-	ulong dev6, dev62;
+	ulong dev6;
 	ulong temp;
 	float cas;
 	int rcd, rp, ras, chan;
 	long *ptr, *ptr2;
 
-	pci_conf_read( 0, 6, 0, 0x40, 4, &dev62);
+	pci_conf_read( 0, 6, 0, 0x40, 4, &dev6);
 	ptr2=(long*)(dev6+0x68);
 
 	/* Read the MMR Base Address & Define the pointer */
